backend/drm/util.c

```diff
@@ -323,10 +324,14 @@ static bool match_obj_(struct match_state *st, size_t skips, size_t score, size_
 	return match_obj_(st, skips, score, replaced, i + 1);
 }
 
-size_t match_obj(size_t num_objs, const uint32_t objs[static restrict num_objs],
-		size_t num_res, const uint32_t res[static restrict num_res],
-		uint32_t out[static restrict num_res]) {
-	uint32_t solution[num_res];
+ssize_t match_obj(size_t num_objs, const uint32_t *objs,
+		size_t num_res, const uint32_t *res,
+		uint32_t *out) {
```

Comment:
You've removed the restrict qualification off of all of the arguments.
Also, VLA arguments are completely safe. I oppose their removal.

Reply:
With `-Wvla`, it will complain about VLA arguments.